src/library-authoring/collections/CollectionDetails.tsx

```diff
@@ -0,0 +1,154 @@
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import { Icon, Stack } from '@openedx/paragon';
+import { useContext, useState } from 'react';
+import classNames from 'classnames';
+
+import { getItemIcon } from '../../generic/block-type-utils';
+import { ToastContext } from '../../generic/toast-context';
+import { BlockTypeLabel, type CollectionHit, useSearchContext } from '../../search-manager';
+import type { ContentLibrary } from '../data/api';
+import { useUpdateCollection } from '../data/apiHooks';
+import HistoryWidget from '../generic/history-widget';
+import messages from './messages';
+
+interface BlockCountProps {
+  count: number,
+  blockType?: string,
+  label: React.ReactNode,
+  className?: string,
+}
+
+const BlockCount = ({
+  count,
+  blockType,
+  label,
+  className,
+}: BlockCountProps) => {
+  const icon = blockType && getItemIcon(blockType);
+  return (
+    <Stack className={classNames('text-center', className)}>
+      <span className="text-muted">{label}</span>
+      <Stack direction="horizontal" gap={1} className="justify-content-center">
+        {icon && <Icon src={icon} size="lg" />}
+        <span>{count}</span>
+      </Stack>
+    </Stack>
+  );
+};
+
+const CollectionStatsWidget = () => {
+  const {
+    blockTypes,
+  } = useSearchContext();
+
+  const blockTypesArray = Object.entries(blockTypes)
+    .map(([blockType, count]) => ({ blockType, count }))
+    .sort((a, b) => b.count - a.count);
+
+  const totalBlocksCount = blockTypesArray.reduce((acc, { count }) => acc + count, 0);
+  const otherBlocks = blockTypesArray.splice(3);
+  const otherBlocksCount = otherBlocks.reduce((acc, { count }) => acc + count, 0);
```

Comment:
Worth adding a comment to explain what's happening here.. I'd also call out a const `numBlockTypesShown` variable, so it's easy to see what it's used for. e.g.:

```suggestion
  // Show the top 3 block type counts individually, and splice the remaining block types together under "Other".
  const numBlockTypesShown = 3;
  const otherBlocks = blockTypesArray.splice(numBlockTypesShown);
  const otherBlocksCount = otherBlocks.reduce((acc, { count }) => acc + count, 0);
```